src/backend/cpu/Event.cpp

```diff
@@ -8,13 +8,59 @@
  ********************************************************/
 
 #include <Event.hpp>
+
+#include <common/err_common.hpp>
+#include <events.hpp>
+#include <platform.hpp>
 #include <queue.hpp>
+#include <af/event.h>
 
 namespace cpu {
 /// \brief Creates a new event and marks it in the queue
-Event make_event(cpu::queue &queue) {
+Event makeEvent(cpu::queue& queue) {
     Event e;
     if (0 == e.create()) { e.mark(queue); }
     return e;
 }
+
+af_event createEvent() {
+    Event* e     = new Event();
+    Event& event = *e;
+    // Ensure that the default queue is initialized
+    getQueue();
+    if (event.create() != 0) {
+        delete e;  // don't leak the event if creation fails
+        AF_ERROR("Could not create event", AF_ERR_RUNTIME);
+    }
+    af_event eventHandle = getHandle(event);
+    markEventOnActiveQueue(eventHandle);
```

Comment:
I understand you were just implementing the make_event function. Perhaps its better to rename this createAndMarkEvent?